src/abstractdataframe/selection.jl

```diff
@@ -69,11 +62,9 @@ function select_transform!(nc::Union{Pair{Int, Pair{ColRename, Symbol}},
         if length(col_idx) == 0
             newdf[!, newname] = [first(last(nc))() for _ in axes(df, 1)]
         else
-            cols = ntuple(i -> _columns(df)[col_idx[i]], length(col_idx))
-            colnames = ntuple(i -> _names(df)[col_idx[i]], length(col_idx))
-            newdf[!, newname] = select_transform_helper(TypeHolder{NamedTuple{colnames,
-                                                                              Tuple{eltype.(cols)...}}}(),
-                                                        cols, first(last(nc)), nrow(df))
+            rowiterator = Tables.rows(Tables.columntable(df[!, col_idx]))
+            newdf[!, newname] = map(first(last(nc)),
+                                    Tables.namedtupleiterator(eltype(rowiterator), rowiterator))
```

Comment:
Do you need both `Tables.rows` and `Tables.namedtupleiterator`?

Reply:
This is what I understood from what @quinnj written (and simpler approaches were either slow or threw an error).

@quinnj - can we simplify it?